File: plugins/shelly/src/main/java/eu/geekhome/shellyplugin/LanInetAddressHelper.java
package eu.geekhome.shellyplugin;

import java.net.*;
import java.util.ArrayList;
import java.util.Enumeration;
import java.util.List;

public class LanInetAddressHelper {

    static List<InetAddress> getIpsInLan() throws SocketException {
        List<InetAddress> addresses = new ArrayList<>();

        for (final Enumeration<NetworkInterface> interfaces =
             NetworkInterface.getNetworkInterfaces();
             interfaces.hasMoreElements(); ) {
            final NetworkInterface cur = interfaces.nextElement();

            if (cur.isLoopback()) {
                continue;
            }

            for (final InterfaceAddress addr : cur.getInterfaceAddresses()) {
                final InetAddress inet_addr = addr.getAddress();

                if (inet_addr instanceof Inet4Address) {
                    addresses.add(inet_addr);
                }
            }
        }

        return addresses;
    }
}

File: plugins/shelly/src/main/java/eu/geekhome/shellyplugin/LanInetAddressHelper.kt
package eu.geekhome.shellyplugin

import java.net.Inet4Address
import java.net.InetAddress
import java.net.NetworkInterface
import java.net.SocketException
import java.util.*

object LanInetAddressHelper {

    @get:Throws(SocketException::class)
    val ipsInLan: List<InetAddress>
        get() {
            val addresses: MutableList<InetAddress> = ArrayList()
            val interfaces = NetworkInterface.getNetworkInterfaces()
            while (interfaces.hasMoreElements()) {
                val cur = interfaces.nextElement()
                if (cur.isLoopback) {
                    continue
                }
                for (addr in cur.interfaceAddresses) {
                    val inet_addr = addr.address
                    if (inet_addr is Inet4Address) {
                        addresses.add(inet_addr)
                    }
                }
            }
            return addresses
        }
}